### PR TITLE
chore: Add deprecation note for customStateCheck

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Add deprecation note for the `customStateCheck` attribute:

**Related issue(s)**

https://github.com/kyma-project/lifecycle-manager/pull/1708
https://github.com/kyma-project/lifecycle-manager/issues/1616

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -344,7 +344,7 @@ You can also define multiple files in `config/samples`, but you must specify the
    - `beta`: (Optional) Determines whether ModuleTemplate must have the beta flag or not. The type is bool.
    - `labels`: (Optional) Additional labels for ModuleTemplate.
    - `annotations`: (Optional) Additional annotations for ModuleTemplate.
-   - `customStateCheck`: (Optional) Specifies custom state checks for the module.
+   - `customStateCheck`: (Optional) DEPRECATED, Specifies custom state checks for the module.
```
Comment (c-pius) [inline README.md:347]: Should we maybe include a date when it was deprecated? Not sure if strictly necessary, but shouldn't harm either.

Comment (Tomasz-Smelcerz-SAP) [reply README.md:347]: @mmitoraj What do you think?

Comment (mmitoraj) [reply README.md:347]: I agree

Comment (mmitoraj) [inline README.md:347]:
```suggestion
   - `customStateCheck`: (Optional) [DEPRECATED] Specifies custom state checks for the module.
   
   > **CAUTION:** This field is deprecated and will be removed on {DATE}. 
```

Comment (mmitoraj) [reply README.md:347]: or we could provide both the deprecation and deletion dates.

```suggestion
   - `customStateCheck`: (Optional) [DEPRECATED], Specifies custom state checks for the module.

   > **CAUTION:** This field was deprecated at the end of July 2024 and will be removed on {DATE}. 
```

Comment (Tomasz-Smelcerz-SAP) [reply README.md:347]: I think we don't know the deletion date/API version. So I'd rather skip this.

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**